src/pages/resultsView/download/DownloadTab.tsx

```diff
@@ -398,9 +399,9 @@ export default class DownloadTab extends React.Component<IDownloadTabProps, {}>
                     {hasValidMutationData(this.mutationData.result!) && this.mutationDownloadControls()}
                     {hasValidData(this.mrnaData.result!) && this.mrnaExprDownloadControls()}
                     {hasValidData(this.proteinData.result!) && this.proteinExprDownloadControls()}
-                    {this.alteredSamplesDownloadControls.component}
-                    {this.unalteredSamplesDownloadControls.component}
-                    {this.sampleMatrixDownloadControls.component}
+                    {this.alteredSamplesDownloadControls(this.alteredCaseAlterationData.result!, this.props.store.virtualStudyParams.result!)}
```

Comment:
lets take this jsx out of the remoteData.  i've never believed that rendering should be done in return of invoke function.    the remoteData should just return data and that should be consumed by component render method.

Reply:
@alisman Sure, so we remove this `handleCloseModal` and move the table in the render function, is that alright?

Reply:
correct.  MobxView is permissable, but I actually still don't really like it.   so for this, lets just put  the code in render method.

Reply:
@alisman Updated, is it looks better?